File: app/src/main/java/com/example/speedtest/GeofenceBroadcastReceiver.java
package com.example.speedtest;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import org.jetbrains.annotations.NotNull;

/**
 * Receiver for geofence transition changes.
 * <p>
 * Receives geofence transition events from Location Services in the form of an Intent containing
 * the transition type and geofence id(s) that triggered the transition. Creates a JobIntentService
 * that will handle the intent in the background.
 */
public class GeofenceBroadcastReceiver extends BroadcastReceiver {

    /**
     * Receives incoming intents.
     *
     * @param context the application context.
     * @param intent  sent by Location Services. This Intent is provided to Location
     *                Services (inside a PendingIntent) when addGeofences() is called.
     */


    static class LocationSpeedTestTask extends SpeedTestTask {
        public LocationSpeedTestTask(@NotNull Context context) {
            super(context);
        }
    }


    @Override
    public void onReceive(Context context, Intent intent) {
        // Enqueues a JobIntentService passing the context and intent as parameters
        Log.v("geofences", "yay");
        new LocationSpeedTestTask(context).execute();

    }
}

File: app/src/main/java/com/example/speedtest/GeofenceBroadcastReceiver.kt
//  # COMP 4521    #  MOHAMAD, Randitya Setyawan    20316273    rsmohamad@ust.hk
//  # COMP 4521    #  IVANOV, Metodi Dimitrov       20314512    mdivanov@connect.ust.hk

package com.example.speedtest

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class GeofenceBroadcastReceiver : BroadcastReceiver() {

    internal class LocationSpeedTestTask(context: Context) : SpeedTestTask(context) {
        override fun onPreExecute() {
            super.onPreExecute()
            contextRef.get()?.let {
                NotificationUtil.sendNotification(
                    it,
                    it.getString(R.string.notification_title),
                    it.getString(R.string.notification_message_location)
                )
            }
        }
    }


    override fun onReceive(context: Context, intent: Intent) {
        // Enqueues a JobIntentService passing the context and intent as parameters
        Log.v("geofences", "yay")
        LocationSpeedTestTask(context).execute()
    }
}